cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaConfigurationDiff.java

```diff
@@ -0,0 +1,210 @@
+            "background.threads",
+            "compression.type",
+            "log.flush.interval.messages",
+            "log.flush.interval.ms",
+            "log.retention.bytes",
+            "log.retention.ms",
+            "log.roll.jitter.ms",
+            "log.roll.ms",
+            "log.segment.bytes",
+            "log.segment.delete.delay.ms",
+            "message.max.bytes",
+            "min.insync.replicas",
+            "num.io.threads",
+            "num.network.threads",
+            "num.recovery.threads.per.data.dir",
+            "num.replica.fetchers",
+            "unclean.leader.election.enable",
+            "log.cleaner.backoff.ms",
+            "log.cleaner.dedupe.buffer.size",
+            "log.cleaner.delete.retention.ms",
+            "log.cleaner.io.buffer.load.factor",
+            "log.cleaner.io.buffer.size",
+            "log.cleaner.io.max.bytes.per.second",
+            "log.cleaner.max.compaction.lag.ms",
+            "log.cleaner.min.cleanable.ratio",
+            "log.cleaner.min.compaction.lag.ms",
+            "log.cleaner.threads",
+            "log.cleanup.policy",
+            "log.index.interval.bytes",
+            "log.index.size.max.bytes",
+            "log.message.timestamp.difference.max.ms",
+            "log.message.timestamp.type",
+            "log.preallocate",
+            "max.connections",
+            "max.connections.per.ip",
+            "max.connections.per.ip.overrides",
+            "log.message.downconversion.enable",
+            "metric.reporters"
+            ));
+
+    public KafkaConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired) {
+        this.current = current;
+        this.desired = desired;
+    }
+
+    private Properties configMap2Properties() {
+        Properties result = new Properties();
+        String desireConf = desired.getData().get("server.config");
+
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(desireConf);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+
+        return validLines;
+    }
+
+    public Set<String> getDiff() {
+        Properties desiredConfig = configMap2Properties();
+        Set<String> result = new HashSet<>();
+        Map.Entry<ConfigResource, Config> currentBrokerConfiguration = current.entrySet().iterator().next();
```

Comment:
Thsi is very confusing for me. Current broker configuration is a map of the actual broker configuration? If so, how do you know the first line gives you broker ID? Do you just expect it to be always on the first line? If so, this is a bit dangerous and you should also explain it in comments. But I think this code is very hard to read and nobody will properly undertsand it in few weeks. It has to be rewritten into something more readable.